docs/specs/add-kubernetes-support.md

```diff
@@ -28,20 +28,62 @@ Start by creating a new pkg for cluster objects, ```pkg/cluster/...```. From
 there, the cluster pkg will hold ```pkg/cluster/cluster.go```,
 ```pkg/cluster/kubernetes.go```, and ```pkg/cluster/openshift.go```.
 
-The file ```pkg/cluster/cluster.go```  will provide an object that will abstract
+The file ```pkg/cluster/cluster.go``` will provide an object that will abstract
 cluster logic.
+```diff
++type COE interface {
++     CreateSandbox()
++     Run(APBActionProvision APBAction, req Context)
++     DestroySandbox()
++}
+
++struct Cluster {
```

Comment:
One question since this was discussed as a `Delegate`: Do we care about runtimes that are **not** kubernetes or openshift? For example: vanilla docker? IMO, the answer is "No, but we should design for the possibility" since we've already had interest expressed from outside the team. With that in mind, are these other runtimes always going to be "Clusters"? I don't think it would be appropriate to call docker that.

My request would be to keep this as the runner's `Delegate`, curious how other folks feel about that. Otherwise, looks great. Thanks for writing this all up @rthallisey.